src/org/zaproxy/zap/extension/frontEndScanner/files/scripts/scripts/scan-jwt-tokens.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * When storage is written to, check if the written value is a JWT token
+ */
+
+frontEndScanner.mailbox.subscribe('storage', function (_, data) {
+  if (data.action === 'set') {
+    verify(data);
+  }
+});
+
+function verify(data) {
+  const potentialToken = data.value;
+  const tokenContent = parseJwt(potentialToken);
+
+  if (tokenContent) {
+    const alert = {
+      confidence: 3,
+      description: 'A JWT token has been found in a storage.',
```

Comment:
Can we tell if its local or session storage? That would be really useful to report. Also should report the key?

Reply:
Actually, it should return the whole JWT to be included in the alert.

Reply:
The [tracker](https://github.com/Pamplemousse/front-end-tracker) currently does not make a difference between the storages... See [this issue](https://github.com/zaproxy/front-end-tracker/issues/1) :)
Should I report the key in the description, or is there a specific field that I could fill?

Reply:
This could go into the evidence.